pkg/ai/iai.go

```diff
@@ -29,5 +40,30 @@ type AIConfiguration struct {
 type AIProvider struct {
 	Name     string `mapstructure:"name"`
 	Model    string `mapstructure:"model"`
-	Password string `mapstructure:"password"`
+	Password string `mapstructure:"password" yaml:"password,omitempty"`
+	BaseURL  string `mapstructure:"baseurl" yaml:"baseurl,omitempty"`
+	Engine   string `mapstructure:"engine" yaml:"engine,omitempty"`
+}
+
+func (p *AIProvider) GetToken() string {
+	return p.Password
+}
+
+func (p *AIProvider) GetModel() string {
+	return p.Model
+}
+
+func (p *AIProvider) GetBaseURL() string {
+	return p.BaseURL
+}
+
+func (p *AIProvider) GetEngine() string {
+	return p.Engine
+}
+
+func NeedPassword(backend string) bool {
```

Comment:
Nothing blocking just a proposal to simplify the method:

```bash
func NeedPassword(backend string) bool {
	return backend != "llama"
}
```

Reply:
good shout, I will resolve it in the PR, I will shortly close this big draft PR..